docs/extras/modules/data_connection/vectorstores/integrations/supabase.mdx

```diff
@@ -45,7 +46,7 @@ begin
   select
     id,
     content,
-    metadata,
+    (metadata || CASE WHEN include_embeddings THEN jsonb_build_object('embedding', (embedding::text)::jsonb) ELSE '{}'::jsonb END) as metadata,
```

Comment:
For simplicity, could we instead always return the embeddings in the query here and filter them out client side?

Reply:
We could, but the embeddings may be very large compared to the rest of the payload, so it will use a lot more network capability if we unnecessarily return it.
A typical document content might be anywhere from 20 to 10000 characters long, while the (OpenAI) embedding of that document always consists of 1536 32-bit floats.
If you insist I can make that simplification, but imho it wouldn't be a very good idea.

Reply:
I'd rather keep it simple for right now - 6kb per doc sounds like a lot but most searches are only a few docs at a time and a few kilobytes of bandwidth isn't going to kill anyone when even your average mobile connection can do that in a fraction of a second

Reply:
I'm worried about this getting tricky to maintain if more edge cases to the metadata field pop up

Reply:
Fair enough. Simplified in [25fd178](https://github.com/langchain-ai/langchainjs/pull/2800/commits/25fd178199307db5cd7d3eaf8e6ba7c3ceb2702d). Note: If you want to run the tests, you will have to update your `match_documents` function in Supabase to the latest simplified version.